Source/JavaScriptCore/runtime/JSPromise.cpp

```diff
@@ -287,7 +287,6 @@ void JSPromise::performPromiseThen(VM& vm, JSGlobalObject* globalObject, JSValue
 
 void JSPromise::rejectPromise(VM& vm, JSGlobalObject* globalObject, JSValue argument)
 {
-    ASSERT(status() == Status::Pending);
```

Comment:
This is actually not correct since this function should be used only after the first resolving call gets done already.